hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java

```diff
@@ -145,6 +147,10 @@ public CallOutcome call() {
                 result = result.withSignerNonce(sender.getNonce());
             }
 
+            if (!result.isSuccess()) {
```

Comment:
I wonder if this if statement can/should be combined with the if statement below?

Reply:
Makes sense but I think we should first verify that this behaviour is the one we are targeting and get back to it to refactor it properly.

Reply:
yes, agreed